crates/ruff_python_parser/tests/snapshots/invalid_syntax@statements__ambiguous_lpar_with_items.py.snap

```diff
@@ -0,0 +1,580 @@
+                                            is_async: false,
+                                        },
+                                    ],
+                                    parenthesized: false,
+                                },
+                            ),
+                            optional_vars: None,
+                        },
+                    ],
+                    body: [
+                        Expr(
+                            StmtExpr {
+                                range: 409..412,
+                                value: EllipsisLiteral(
+                                    ExprEllipsisLiteral {
+                                        range: 409..412,
+                                    },
+                                ),
+                            },
+                        ),
+                    ],
+                },
+            ),
+        ],
+    },
+)
+```
+## Errors
+
+  |
+2 | # These cases should raise the correct syntax error and recover properly.
+3 | 
+4 | with (item1, item2),: ...
+  |                     ^ Syntax Error: Expected an expression
+5 | with (item1, item2), as f: ...
+6 | with (item1, item2), item3,: ...
+  |
+
+
+  |
+4 | with (item1, item2),: ...
+5 | with (item1, item2), as f: ...
+  |                      ^^ Syntax Error: Expected an expression
+6 | with (item1, item2), item3,: ...
+7 | with (*item): ...
+  |
+
+
+  |
+4 | with (item1, item2),: ...
+5 | with (item1, item2), as f: ...
+6 | with (item1, item2), item3,: ...
+  |                           ^ Syntax Error: Trailing comma not allowed
+7 | with (*item): ...
+8 | with (item := 10 as f): ...
+  |
+
+
+  |
+5 | with (item1, item2), as f: ...
+6 | with (item1, item2), item3,: ...
+7 | with (*item): ...
+  |       ^^^^^ Syntax Error: cannot use starred expression here
+8 | with (item := 10 as f): ...
+9 | with (item1, item2 := 10 as f): ...
+  |
+
+
+   |
+ 6 | with (item1, item2), item3,: ...
+ 7 | with (*item): ...
+ 8 | with (item := 10 as f): ...
+   |       ^^^^^^^^^^ Syntax Error: unparenthesized named expression cannot be used here
+ 9 | with (item1, item2 := 10 as f): ...
+10 | with (x for x in range(10), item): ...
+   |
+
+
+   |
+ 7 | with (*item): ...
+ 8 | with (item := 10 as f): ...
+ 9 | with (item1, item2 := 10 as f): ...
+   |              ^^^^^^^^^^^ Syntax Error: unparenthesized named expression cannot be used here
+10 | with (x for x in range(10), item): ...
+11 | with (item, x for x in range(10)): ...
+   |
+
+
+   |
+ 8 | with (item := 10 as f): ...
+ 9 | with (item1, item2 := 10 as f): ...
+10 | with (x for x in range(10), item): ...
+   |                           ^ Syntax Error: expected Rpar, found Comma
+11 | with (item, x for x in range(10)): ...
+   |
+
+
+   |
+ 8 | with (item := 10 as f): ...
+ 9 | with (item1, item2 := 10 as f): ...
+10 | with (x for x in range(10), item): ...
+   |                                 ^ Syntax Error: expected Comma, found Rpar
```

Comment:
I don't think I understand why it expects a comma after `item`. Python errors after the first comma instead. 

Is it because we try to parse this as a tuple? But even adding a trailing comma doesn't produce valid syntax. I think we need to improve this error message to ensure it's a valid suggestion.

Reply:
Fixed in https://github.com/astral-sh/ruff/pull/10490